File: smelly/duplicate-assert/java-junit/TestInstrumentation_testSamplers.java
public class TestInstrumentation extends TestCase {
    public void testCron() throws Exception {
        Instrumentation.Cron cron = new Instrumentation.Cron();
        long start = System.currentTimeMillis();
        assertEquals("", 0, cron.getStart(), TOLERANCE);
        assertEquals("", 0, cron.getEnd(), TOLERANCE);
        assertEquals(cron.getStart(), cron.getEnd());
        assertEquals(0, cron.getOwn());
        assertEquals(0, cron.getTotal());

        cron.start();
        long s = System.currentTimeMillis();
        Thread.sleep(INTERVAL);
        cron.stop();
        long realOwnDelay = System.currentTimeMillis() - s;
        long now = System.currentTimeMillis();
        assertEquals("", start, cron.getStart(), TOLERANCE);
        assertEquals("", now, cron.getEnd(), TOLERANCE);
        assertEquals("", INTERVAL, cron.getTotal(), TOLERANCE);
        assertEquals("", INTERVAL, cron.getOwn(), TOLERANCE);
        assertEquals("", cron.getTotal(), cron.getOwn(), TOLERANCE);

        long realTotalDelay = System.currentTimeMillis() - s;
        s = System.currentTimeMillis();
        Thread.sleep(INTERVAL);

        cron.start();

        realTotalDelay += System.currentTimeMillis() - s;

        s = System.currentTimeMillis();
        Thread.sleep(INTERVAL);
        cron.stop();
        now = System.currentTimeMillis();

        realTotalDelay += System.currentTimeMillis() - s;
        realOwnDelay += System.currentTimeMillis() - s;

        assertEquals("", start, cron.getStart(), TOLERANCE);
        assertEquals("", now, cron.getEnd(), TOLERANCE);
        assertEquals("", realTotalDelay, cron.getTotal(), TOLERANCE);
        assertEquals("", realOwnDelay, cron.getOwn(), TOLERANCE);
    }
}

File: smelly/duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/TestInstrumentation_testSamplers.cs
public class TestInstrumentation
{
    [Fact]
    public void TestCron()
    {
        var cron = new Instrumentation.Cron();
        long start = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        Assert.Equal(0, cron.GetStart(), TOLERANCE);
        Assert.Equal(0, cron.GetEnd(), TOLERANCE);
        Assert.Equal(cron.GetStart(), cron.GetEnd());
        Assert.Equal(0, cron.GetOwn());
        Assert.Equal(0, cron.GetTotal());

        cron.Start();
        long s = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        Thread.Sleep(INTERVAL);
        cron.Stop();
        long realOwnDelay = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - s;
        long now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        Assert.Equal(start, cron.GetStart(), TOLERANCE);
        Assert.Equal(now, cron.GetEnd(), TOLERANCE);
        Assert.Equal(INTERVAL, cron.GetTotal(), TOLERANCE);
        Assert.Equal(INTERVAL, cron.GetOwn(), TOLERANCE);
        Assert.Equal(cron.GetTotal(), cron.GetOwn(), TOLERANCE);

        long realTotalDelay = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - s;
        s = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        Thread.Sleep(INTERVAL);

        cron.Start();

        realTotalDelay += DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - s;

        s = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
        Thread.Sleep(INTERVAL);
        cron.Stop();
        now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();

        realTotalDelay += DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - s;
        realOwnDelay += DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - s;

        Assert.Equal(start, cron.GetStart(), TOLERANCE);
        Assert.Equal(now, cron.GetEnd(), TOLERANCE);
        Assert.Equal(realTotalDelay, cron.GetTotal(), TOLERANCE);
        Assert.Equal(realOwnDelay, cron.GetOwn(), TOLERANCE);
    }
}
